.github/workflows/release.yml

```diff
@@ -48,8 +50,9 @@ jobs:
         uses: actions/cache@v3
         with:
           path: ~/bazel-disk-cache
-          key: ${{ runner.os }}-${{ runner.arch }}-bazel-disk-cache
-
+          key: bazel-disk-cache-${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE') }}
```

Comment:
Not for this change, but would we be better off with separate caches for release (`opt`) and test (`fastbuild`) builds? Alternatively, is using the same build configuration for test and release feasible? Ideally we'd have both in CI if they are different.

Reply:
Sounds good, but leads to the question that if test and release take about the same time, can we align the test and release configs so CI is testing what is released (currently it's `fastbuild` for test and `opt` for release)? Ideally, it would be better test test all compilation modes / settings including dbg, but that's not viable with the available resources.

Reply:
In local testing opt builds are only slightly slower than fastbuild (perhaps 1.5x or even less). The other reason the release job is taking about the same time is that it only has to build the workerd binary instead of needing to build and test everything.
Based on the above I'd still expect a CI slowdown from running release builds, but the caches to be downloaded would be smaller and tests run more quickly so the difference might be small enough to be acceptable.
On the other hand a separate release/opt cache would be smaller (release builds are smaller, plus we're only building workerd itself), perhaps small enough to be worth having.
I guess both approaches are worth exploring, but perhaps something to do later on.